src/Gui/Selection.cpp

```diff
@@ -420,6 +422,18 @@ void SelectionSingleton::notify(SelectionChanges &&Chng)
             notify = true;
         }
         if(notify) {
+            // Notify the view provider of the object as well.
+            App::Document *doc = App::GetApplication().getDocument(msg.pDocName);
+            if(doc) {
+                App::DocumentObject *obj = doc->getObject(msg.pObjectName);
+                if(obj) {
+                    auto vp = Application::Instance->getViewProvider(obj);
+                    if (vp) {
+                        vp->onSelectionChanged(msg);
+                    }
+                }
+            }
+

```

Comment:
FYI you can initialize variables in `if` statements:
```suggestion
            // Notify the view provider of the object as well.
            if(const auto* doc = App::GetApplication().getDocument(msg.pDocName)) {
                if(const auto* obj = doc->getObject(msg.pObjectName)) {
                    if (auto* vp = Application::Instance->getViewProvider(obj) {
                        vp->onSelectionChanged(msg);
                    }
                }
            }
```